source/tests/system/nixnet_utilities.h

```diff
@@ -0,0 +1,193 @@
+
+#define EXPECT_SUCCESS(response_)    \
+  ([](const auto& response) {              \
+    EXPECT_EQ(0, response.status()); \
+    return response;                 \
+  })(response_)
+
+#define EXPECT_XNET_ERROR(error, response) \
+  if (1) {                                 \
+    EXPECT_EQ(error, (response).status()); \
+  }
+
+namespace nixnet_utilities {
+
+namespace client = nixnet_grpc::experimental::client;
+namespace pb = google::protobuf;
+using namespace nixnet_grpc;
+
+typedef pb::uint32 u32;
+
+GetPropertyResponse get_property(const client::StubPtr& stub, const nidevice_grpc::Session& session_ref, const client::simple_variant<nixnet_grpc::Property, pb::uint32>& property_id)
+{
+  ::grpc::ClientContext context;
+
+  auto request = GetPropertyRequest{};
+  request.mutable_session()->CopyFrom(session_ref);
+  const auto property_id_ptr = property_id.get_if<nixnet_grpc::Property>();
+  const auto property_id_raw_ptr = property_id.get_if<pb::uint32>();
+  if (property_id_ptr) {
+    request.set_property_id(*property_id_ptr);
+  }
+  else if (property_id_raw_ptr) {
+    request.set_property_id_raw(*property_id_raw_ptr);
+  }
+
+  auto response = GetPropertyResponse{};
+
+  client::raise_if_error(
+      stub->GetProperty(&context, request, &response));
+
+  return response;
+}
+
+void _set_property_value(SetPropertyRequest& request, const pb::uint32& value)
+{
+  request.set_u32_scalar(value);
+}
+
+void _set_property_value(SetPropertyRequest& request, const bool& value)
+{
+  request.set_bool_scalar(value);
+}
+
+void _set_property_value(SetPropertyRequest& request, const std::string& value)
+{
+  request.set_str(value);
+}
+
+void _set_property_value(SetPropertyRequest& request, const pb::uint64& value)
+{
+  request.set_u64_scalar(value);
+}
+
+void _set_property_value(SetPropertyRequest& request, const pb::int32& value)
+{
+  request.set_i32_scalar(value);
+}
+
+void _set_property_value(SetPropertyRequest& request, const double& value)
+{
+  request.set_f64_scalar(value);
+}
+
+void _set_property_value(SetPropertyRequest& request, const std::vector<std::string>& value)
+{
+  // TODO: set_string_array expects a single string?!?
+  request.set_string_array(value[0]);
+}
+
+void _set_property_value(SetPropertyRequest& request, const std::vector<pb::uint32>& value)
+{
+  request.mutable_u32_array()->clear_u32_array();
+  request.mutable_u32_array()->mutable_u32_array()->Add(value.begin(), value.end());
+}
+
+void _set_property_value(SetPropertyRequest& request, const nidevice_grpc::Session& value)
+{
+  request.mutable_db_ref()->CopyFrom(value);
+}
+
+void _set_property_value(SetPropertyRequest& request, const std::vector<nidevice_grpc::Session>& value)
+{
+  request.mutable_db_ref_array()->clear_db_ref();
+  request.mutable_db_ref_array()->mutable_db_ref()->Reserve(static_cast<int>(value.size()));
+  for (int i = 0; i < value.size(); ++i) {
+    request.mutable_db_ref_array()->mutable_db_ref()->at(i) = value[i];
+  }
+}
+
+/* TODO
```

Comment:
Is there a reason this method's commented out here but not in the other PR adding this file? https://github.com/ni/grpc-device/pull/643/files#diff-a1f1455debc1cbf60ed5309c004f84853b90fa1ed29c012236e98af36ea44c70

Reply:
(Technically, it's because it was commented out in my branch where I added this function, and this PR didn't need an implementation for it. At the time I committed my branch, EptRxFilter wasn't yet committed to main. The differences will need to be resolved at some point, either now or when merging.)

Reply:
Yes, and both mine and Palash's changes are introducing the nixnet_utilities.h file. So I am waiting for his PR to go through and update my branch with his changes and see if anything else is needed to be added in or removed from there